apps/server/src/modules/tool/uc/school-external-tool.uc.spec.ts

```diff
@@ -0,0 +1,144 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { schoolExternalToolDOFactory } from '@shared/testing/factory/domainobject/school-external-tool.factory';
+import { SchoolExternalToolDO } from '@shared/domain/domainobject/external-tool/school-external-tool.do';
+import { Actions, Permission, User } from '@shared/domain';
+import { setupEntities, userFactory } from '@shared/testing';
+import { MikroORM } from '@mikro-orm/core';
+import { SchoolDO } from '@shared/domain/domainobject/school.do';
+import { schoolDOFactory } from '@shared/testing/factory/domainobject/school.factory';
+import { SchoolExternalToolUc } from './school-external-tool.uc';
+import { AuthorizationService } from '../../authorization';
+import { SchoolExternalToolService } from '../service/school-external-tool.service';
+import { SchoolService } from '../../school';
+
+describe('SchoolExternalToolUc', () => {
+	let module: TestingModule;
+	let orm: MikroORM;
+	let uc: SchoolExternalToolUc;
+
+	let authorizationService: DeepMocked<AuthorizationService>;
+	let schoolExternalToolService: DeepMocked<SchoolExternalToolService>;
+	let schoolService: DeepMocked<SchoolService>;
+
+	beforeAll(async () => {
+		orm = await setupEntities();
+		module = await Test.createTestingModule({
+			providers: [
+				SchoolExternalToolUc,
+				{
+					provide: AuthorizationService,
+					useValue: createMock<AuthorizationService>(),
+				},
+				{
+					provide: SchoolExternalToolService,
+					useValue: createMock<SchoolExternalToolService>(),
+				},
+				{
+					provide: SchoolService,
+					useValue: createMock<SchoolService>(),
+				},
+			],
+		}).compile();
+
+		uc = module.get(SchoolExternalToolUc);
+		authorizationService = module.get(AuthorizationService);
+		schoolExternalToolService = module.get(SchoolExternalToolService);
+		schoolService = module.get(SchoolService);
+	});
+
+	afterAll(async () => {
+		await module.close();
+		await orm.close();
```

Comment:
Do we really need to call `orm.close()`? The ORM should be closed automatically by a hook when the module is closed (AFAIK).

Reply:
MongoMemoryDatabaseModule.onModuleDestroy will not be called in tests. Because we are not sure i think it is better  to close it